### PR TITLE
USWDS - Input: Add error state story

### LINKED ISSUE #5991: USWDS - Feature: Create Input error state Storybook preview
# Description

Create Storybook preview of Input in error state to be used for site guidance

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

Created error state Storybook controls and preview to showcase error styles.

## Breaking change

This is _not_ a breaking change

## Related issue

Closes #5991

## Related pull requests

_No changelog created. Not a user-facing change._

## Preview link

[Text input error story →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-input-error-story/?path=/story/components-form-inputs-text-input--error)

## Problem statement

We need stories to showcase error states on the design system website.

## Solution

Create an error state storybook preview for text input to match existing controls.

## Major changes

- New option for state control for text input component
- New error state input story
- Conditional logic wraps each text input and it's relevant labels in `usa-form-group usa-form-group--error` divs.

## Testing and review

1. Visit text input story
2. Confirm controls work as expected
3. Confirm the error input has expected markup
4. Confirm error state is displaying correctly
5. Confirm there are no regressions in other states
6. Confirm pattern matches existing variants / components

### CODE REVIEW

File: packages/usa-input/src/usa-input.twig [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+{% if state == "error" %}<div class="usa-form-group usa-form-group--error">{%- endif %}
+  <label class="usa-label{% if state == "error" %} usa-label--error{%- endif %}" for="input-type-text">Text input label</label>
+  {% if state == "error" %} <span class="usa-error-message" id="input-error-message" role="alert">Helpful error message</span>{%- endif %}
+  <input class="usa-input
+    {% if state == "focus" %} usa-focus{%- endif%}
+    {% if state == "error" %} usa-input--error{%- endif%}
+    {% if state == "success" %} usa-input--success{%- endif%}"
+    {% if state == "disabled" %} disabled{%- endif%}
+    {% if state == "aria-disabled" %} aria-disabled="true"{%- endif%}
+    id="input-type-text" name="input-type-text" type="text">
```
Comment (amyleadem) [inline packages/usa-input/src/usa-input.twig:10]: **issue:** This is missing the `aria-describedby` attribute that ties it to the error message
```suggestion
    id="input-type-text" name="input-type-text" type="text"
    aria-describedby="input-error-message">
```

**Review by amyleadem — CHANGES_REQUESTED** [review]
Thanks for getting this prepped so quickly, @mahoneycm. I had a few small notes in the comments below. Let me know if you have any questions.

File: packages/usa-input/src/usa-input.stories.js [hunk 2]
```diff
@@ -38,6 +38,11 @@ AriaDisabled.args = {
   state: "aria-disabled",
 };
 
+export const Error = Template.bind({});
+Error.args = {
```
[the remark this answers is not in the document]
Comment (mahoneycm) [reply packages/usa-input/src/usa-input.stories.js:42]: **question:** If we add "state" here, should we add it to `Disabled` as well (i.e. Disabled State)?  At first I was thinking they should match but maybe it's fine if they don't 🤔

File: packages/usa-input/src/usa-input.twig (hunk 1 above)
Comment (mahoneycm) [reply packages/usa-input/src/usa-input.twig:10]: Thanks for catching this! Resolved in [af4c99f](https://github.com/uswds/uswds/pull/5996/commits/af4c99f54c2b2eba2fa61748394e45eec8e81cd4)

File: packages/usa-input/src/usa-input.twig [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
+{% if state == "error" %}</div>{%- endif %}
+{% if state == "error" %}<div class="usa-form-group usa-form-group--error">{%- endif %}
```
[the remark this answers is not in the document]
Comment (mahoneycm) [reply packages/usa-input/src/usa-input.twig]: I like this! Resolved in [af4c99f](https://github.com/uswds/uswds/pull/5996/commits/af4c99f54c2b2eba2fa61748394e45eec8e81cd4)

File: packages/usa-input/src/usa-input.stories.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (amyleadem) [reply packages/usa-input/src/usa-input.stories.js:42]: Hmm...Good point. I just noticed that the [file input error story](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/?path=/story/components-form-inputs-file-input--error) does not include the word "state" either. Let's table this suggestion until we start building out all of the error state stories. It looks like we already have enough precedent to leave "state" off of the stories for now.

**Review by amyleadem — APPROVED** [review]
Looks good to me! Thanks @mahoneycm

**Review by mejiaj — APPROVED** [review]
LGTM, I can see all form error states.

**Review by thisisdano — APPROVED** [review]

**Pull request merged**